### PR TITLE
Separate success and failure flows in asoctl resource importing

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
## What this PR does

Separates the handling of successful resource imports from failures, clearing the way for an upcoming change to the way resources are named. 

## How does this PR make you feel?

gif

### CODE REVIEW

File: v2/cmd/asoctl/pkg/importresources/importable_resource.go [hunk 1]
```diff
@@ -36,18 +36,10 @@ type ImportableResource interface {
...
 	Import(
-		ctx context.Context,
-		log logr.Logger,
-	) (ImportedResource, error)
-
-	// FindChildren returns any child resources that need to be imported.
-	// ctx allows for cancellation of the import.
-	// Returns any additional resources that also need to be imported, as well as any errors that occur.
-	// Partial success is allowed, but the caller should be notified of any errors.
-	FindChildren(
```
Comment (theunrepentantgeek) [inline v2/cmd/asoctl/pkg/importresources/importable_resource.go:47]: Having `FindChildren()` as a separate method introduced some temporal coupling in the interface, as you had to remember to call `Import()` first. I've collapsed both into a single call.

File: v2/cmd/asoctl/pkg/importresources/imported_resource.go [hunk 2]
```diff
@@ -26,7 +26,4 @@ type ImportedResource interface {
 
 	// Resource returns the actual resource that has been imported.
 	Resource() genruntime.MetaObject
-
-	// Error returns any error that occurred during the import.
-	Error() error
```
Comment (theunrepentantgeek) [inline v2/cmd/asoctl/pkg/importresources/imported_resource.go:31]: This was unused.

File: v2/cmd/asoctl/pkg/importresources/resource_importer.go [hunk 3]
```diff
@@ -37,9 +37,8 @@ type ResourceImporterOptions struct {
 }
 
 type ImportResourceResult struct {
-	resource ImportedResource
-	pending  []ImportableResource
-	err      error
```
Comment (theunrepentantgeek) [inline v2/cmd/asoctl/pkg/importresources/resource_importer.go:42]: Now using a separate channel to return failures, which gives this struct a firmer purpose.

File: v2/cmd/asoctl/pkg/importresources/resource_importer_report.go [hunk 4]
```diff
@@ -47,21 +48,26 @@ func newResourceImportReport() *resourceImportReport {
 }
 
 // AddSuccessfulImport adds a successful import to the report
-func (r *resourceImportReport) AddSuccessfulImport(imported ImportedResource) {
+func (r *resourceImportReport) AddSuccessfulImport(
```
Comment (theunrepentantgeek) [inline v2/cmd/asoctl/pkg/importresources/resource_importer_report.go:51]: The report is used to collate totals, so it doesn't need the full resource, just its GroupKind.

**Review by matthchr — APPROVED** [review]

**Pull request merged**